hw/mcu/dialog/da1469x/src/hal_flash.c

```diff
@@ -395,35 +396,121 @@ da1469x_hff_sector_info(const struct hal_flash *dev, int idx,
     return 0;
 }
 
-#if MYNEWT_VAL(MCU_QSPIC_APP_CFG)
+#if MYNEWT_VAL(RAM_RESIDENT)
+struct qspi_qe_config {
+    uint8_t length;
+    uint8_t bytes[2];
+};
+
+struct qspi_flash_config {
+    uint32_t id;
+    uint32_t cmda;
+    uint32_t cmdb;
+    struct qspi_qe_config qe;
+};
+
+static const struct qspi_flash_config rdids[] =
+{
+    { .id = 0xc86015,                       /* gigaflash */
```

Comment:
I'll fix that.  As for moving this to BSP, that's fine.  I am unsure about the possibility of the QE enable stuff changing between products in the same manufacturer.  I imagine it could be possible, but it might suffice to just use the vendor id.  The alternative is just compiling a list of complete RDIDs (vendor/product), which is what other projects like linux do.

Reply:
Updated commit to implement the suggestions and to fix the gigadevice comment.